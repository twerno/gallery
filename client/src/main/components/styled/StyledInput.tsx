import styled from 'styled-components';

export default styled.input`
    padding: 9px 18px;
    width: 100%;
    border: .75px solid #cecece;
    font-size: 1rem;

    &:focus {
        outline-color: rgb(103, 156, 226);
    }

`;
