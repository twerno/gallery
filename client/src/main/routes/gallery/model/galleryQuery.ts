export interface IGalleryUrlQuery {
    q: string | undefined;
}

export const GalleryQueryHelper = {

    buildEmpty,
    buildFromLocationSearch

}

function buildEmpty(): IGalleryUrlQuery {
    return { q: undefined };
}

function buildFromLocationSearch(search: string): IGalleryUrlQuery {
    const searchParams = new URLSearchParams(search);
    return {
        q: searchParams.get('q') || undefined
    };
}