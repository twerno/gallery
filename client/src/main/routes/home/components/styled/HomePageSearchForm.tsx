import { useFormik } from 'formik';
import { breakpoints } from 'main/globalStyles';
import styled from 'styled-components';
import React from 'react';
import StyledInput from 'main/components/styled/StyledInput';
import StyledButton from 'main/components/styled/StyledButton';
import { IGalleryUrlQuery } from 'main/routes/gallery/model/galleryQuery';

const StyledForm = styled.form`
        width: 50%;
        display: flex;
        justify-content: center;
        align-items: center;
        flex-direction: column;

        & input {
            margin-bottom: 1rem;
        }

        @media only screen and (max-width: ${breakpoints.lg}) {

            & {
                width: 80%;
            }
        }

        @media only screen and (max-width: ${breakpoints.sm}) {

            & {
                width: 90%;
            }
        }
    `;


interface IHomePageSearchFormProps {
    initialValues: IGalleryUrlQuery;
    onSubmit: (values: IGalleryUrlQuery) => Promise<void>
}

export default ({ initialValues, onSubmit }: IHomePageSearchFormProps) => {

    const { handleChange, handleSubmit, values, isSubmitting } = useFormik({
        initialValues,
        onSubmit
    });

    return (
        <StyledForm onSubmit={handleSubmit}>

            <StyledInput
                id="q"
                name="q"
                aria-label="Search"
                type="text"
                onChange={handleChange}
                value={values.q || ''}
            />

            <StyledButton
                mode='standalone'
                type="submit"
                disabled={isSubmitting}
            >
                Search
            </StyledButton>

        </StyledForm>
    );
}
