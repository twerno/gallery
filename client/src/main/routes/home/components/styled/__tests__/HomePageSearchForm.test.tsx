import { wait } from '@testing-library/react';
import { mount } from 'enzyme';
import StyledInput from 'main/components/styled/StyledInput';
import * as React from 'react';
import { componentBasicTests } from "test/componentBasicTests";
import HomePageSearchForm from '../HomePageSearchForm';
import { IGalleryUrlQuery } from 'main/routes/gallery/model/galleryQuery';

describe("HomePageSearchForm", () => {

    componentBasicTests(
        "HomePageSearchForm",
        <HomePageSearchForm initialValues={{ q: '' }} onSubmit={async () => { }} />
    );

    test("submit form", async () => {
        const queryStr = 'cats';
        let submitedValues: IGalleryUrlQuery | undefined;

        const onSubmit = async (values: IGalleryUrlQuery) => { submitedValues = values }

        const wrapper = mount(
            <HomePageSearchForm
                initialValues={{ q: '' }}
                onSubmit={onSubmit}
            />
        );

        const input = wrapper.find(StyledInput);
        expect(input.exists()).toBe(true);
        await wait(() => input.simulate('change', { target: { name: "q", value: queryStr } }));

        const form = wrapper.find('form');
        expect(form.exists()).toBe(true);
        await wait(() => form.simulate('submit', {}));

        expect(submitedValues).toEqual<IGalleryUrlQuery>({ q: queryStr });
    })
});