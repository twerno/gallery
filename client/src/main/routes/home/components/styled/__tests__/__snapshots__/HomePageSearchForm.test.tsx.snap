// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageSearchForm [HomePageSearchForm] Component basic tests matches the snapshot 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    aria-label="Search"
    id="q"
    name="q"
    onChange={[Function]}
    type="text"
    value=""
  />
  <styled.button
    disabled={false}
    mode="standalone"
    type="submit"
  >
    Search
  </styled.button>
</styled.form>
`;
